pytest_fixtures/satellite_auth.py

```diff
@@ -0,0 +1,167 @@
+import copy
+
+from airgun.session import Session
+from nailgun import entities
+
+from robottelo.config import settings
+from robottelo.constants import AUDIENCE_MAPPER
+from robottelo.constants import GROUP_MEMBERSHIP_MAPPER
+from robottelo.constants import LDAP_ATTR
+from robottelo.constants import LDAP_SERVER_TYPE
+from robottelo.datafactory import gen_string
+from robottelo.decorators import fixture
+from robottelo.rhsso_utils import create_mapper
+from robottelo.rhsso_utils import get_rhsso_client_id
+from robottelo.rhsso_utils import run_command
+
+
+@fixture(scope='function')
+def auth_source(ldap_data, module_org, module_loc):
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ldap_data['ldap_user_name'],
+        account_password=ldap_data['ldap_user_passwd'],
+        base_dn=ldap_data['base_dn'],
+        groups_base=ldap_data['group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login_ad'],
+        server_type=LDAP_SERVER_TYPE['API']['ad'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ldap_data['ldap_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture(scope='function')
+def auth_source_ipa(ipa_data, module_org, module_loc):
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ipa_data['ldap_ipa_user_name'],
+        account_password=ipa_data['ldap_ipa_user_passwd'],
+        base_dn=ipa_data['ipa_base_dn'],
+        groups_base=ipa_data['ipa_group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login'],
+        server_type=LDAP_SERVER_TYPE['API']['ipa'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ipa_data['ldap_ipa_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture()
+def ldap_user_name(ldap_data, test_name):
+    """Add LDAP user to satellite by logging in, return username to test and delete the user (if
+    still exists) when test finishes.
+    """
+    with Session(test_name, ldap_data['ldap_user_name'], ldap_data['ldap_user_passwd']):
+        pass
+    yield ldap_data['ldap_user_name']
+    users = entities.User().search(
+        query={'search': 'login="{}"'.format(ldap_data['ldap_user_name'])}
+    )
+    if users:
+        users[0].delete()
```

Comment:
If it's creating and deleting for each test then whats idea here for implementing it as a fixture? a normal function would have done same for you.

I understand that now this can be used other test modules as well, but these type of utilities we used to right as helper functions.

Need to discuss on whether we want to draw the line between helpers and fixtures.

@mshriver Thoughts ?